### PR TITLE
fix: Hide country zip code from Stripe form to avoid duplicate entry

### PR DESCRIPTION
**Pull request by julianajlk**
[REV-3064](https://2u-internal.atlassian.net/browse/REV-3064).

We ask the user for the country and zip code on the address form. We also ask for the same in Stripe's credit card fields.
Stripe saves the credit card country and zip code values, but we want to keep the country and zip code from the address form for SDN check, plus not ask the user for this information twice.

**Before:**
Screen Shot 2022-10-25 at 11 19 29 AM

**After (edX branding not applied):**
Screen Shot 2022-10-25 at 11 47 46 AM

**Zip Code disclaimer:**
For Stripe, zip code is not required to process payments. However, on the frontend validation, it complains if it's undefined. I added a `''` default if `undefined`. Same goes for the unit/line2 address field.
Example of a payment done with Stripe without a zip code:
https://dashboard.stripe.com/test/payments/pi_3LwYtsIadiFyUl1x0PA56G9q
It does not have "Zip check" under Payment method but that does not block payment.

### CODE REVIEW

File: src/payment/checkout/payment-form/StripePaymentForm.jsx [hunk 1]
```diff
@@ -133,8 +133,8 @@ function StripePaymentForm({
               city,
               country,
               line1: address,
-              line2: unit,
-              postal_code: postalCode,
+              line2: unit || '',
+              postal_code: postalCode || '',
```
Comment (julianajlk) [inline src/payment/checkout/payment-form/StripePaymentForm.jsx:137]: line2 and postal code are optional, but Stripe complains about it if we send `undefined`

File: src/payment/checkout/Checkout.jsx [hunk 2]
```diff
@@ -72,14 +72,12 @@ class Checkout extends React.Component {
     );
   };
 
-  handleSubmitStripe = () => {
-    // TODO: We'll want to submit formData here in the next iteration
-    console.log('[Project Zebra] handleSubmitStripe called');
-    this.props.submitPayment({ method: 'stripe' });
-  };
+  // handleSubmitStripe = () => {
```
Comment (wdrussell2015) [inline src/payment/checkout/Checkout.jsx:75]: nit: is there a reason to keep this?

**Review by wdrussell2015 — APPROVED** [review]
looks good to me

File: src/payment/checkout/Checkout.jsx (hunk 2 above)
Comment (julianajlk) [reply src/payment/checkout/Checkout.jsx:75]: It was added when I thought we were going to refactor _soon_ 😅 
I'll remove it, thanks Wes!

**Pull request merged**